src/Unread.js

```diff
@@ -44,6 +45,7 @@ module.exports = {
         // we have and the read receipt. We could fetch more history to try & find out,
         // but currently we just guess.
 
+        const shouldHideEvent = new ShouldHideEvent();
```

Comment:
Design: This shouldn't be defined as an object instance. There's no benefit to doing so as we don't have any internal state to maintain. When is it safe to get rid of this instance? Can I only make it once? Do I make one for every event to check or can I reuse it? It would be clearer if this were just a pure exported function:
```js
import shouldHideEvent from './ShouldHideEvent';

if (shouldHideEvent(ev)) {
  // ...
}
```

You can still have `_isLeaveOrJoin` as a private function (I would actually advise you did this as a proper private function rather than exported with an `_`, it's not really clear why you're doing that).

Bonus points for injecting `UserSettingsStore.getSyncedSettings()` in the function so we can test the logic you're adding. So the final thing looks like:

```js
// src/ShouldHideEvent.js

function _isLeaveOrJoin(ev) {
  // ...
}

export default function(ev, syncedSettings) {
  // Hide redacted events
  if (syncedSettings['hideRedactions'] && ev.isRedacted()) return true;
  if (syncedSettings['hideJoinLeaves'] && _isLeaveOrJoin(ev)) return true;
  return false;
}
```
which can be called/tested like:
```js
import shouldHideEvent from './ShouldHideEvent';

let syncedSettings = UserSettingsStore.getSyncedSettings();
syncedSettings = { or_some: "mocked values maybe" };

if (shouldHideEvent(ev, syncedSettings)) {
  // ...
}
```